beacon-chain/p2p/options.go

```diff
@@ -0,0 +1,51 @@
+package p2p
+
+import (
+	"crypto/ecdsa"
+	"fmt"
+	"net"
+
+	"github.com/libp2p/go-libp2p"
+	"github.com/libp2p/go-libp2p-core/peer"
+	ma "github.com/multiformats/go-multiaddr"
+	"github.com/prysmaticlabs/prysm/shared/iputils"
+)
+
+// buildOptions for the libp2p host.
+func buildOptions(cfg *Config) ([]libp2p.Option, net.IP, *ecdsa.PrivateKey) {
+	ip, err := iputils.ExternalIPv4()
```

Comment:
What about `cfg.HostAddress`? This is generally used to advertise the address that should be used for peer connections

Reply:
ok refactored it a bit, to take in the host address as default